### PR TITLE
[LIT] Add support for `%basename_s` to get base name of source file

### PR DESCRIPTION
**Pull request by jurahul**
Add support for `%basename_s` pattern in the RUN commands to get the base name of the source file, and adopt it in a TableGen LIT test.

### CODE REVIEW

**Review by ilovepi — COMMENTED** [review]
This seems mostly fine, but we should probably have a test under llvm/utils/lit/tests (https://github.com/llvm/llvm-project/tree/main/llvm/utils/lit/tests), rather than just using it in a single tablegen test. Would you mind moving the test there?

**Comment by jurahul** [comment]
> This seems mostly fine, but we should probably have a test under llvm/utils/lit/tests (https://github.com/llvm/llvm-project/tree/main/llvm/utils/lit/tests), rather than just using it in a single tablegen test. Would you mind moving the test there?

Thanks. I'll add a new test instead of moving the TableGen test.

File: llvm/utils/lit/tests/substitutions.py
Comment (ilovepi) [inline llvm/utils/lit/tests/substitutions.py:5]: For a file w/ a single check, I think `CHECK` is probably a better choice over a specific prefix. If you plan to follow up w/ more tests, then using the prefix makes sense though, so I'll leave that up to you to decide on the better approach.

**Review by ilovepi — APPROVED** [review]
LGTM, modulo one small comment and premerge checks are passing.

File: llvm/utils/lit/tests/substitutions.py
Comment (jurahul) [reply llvm/utils/lit/tests/substitutions.py]: Thanks, changed it back to just CHECK. I did look at adding more checks, but some things like full path, it wasn't clear how to check that. So, restricting the test to just the new substitutions being added. if/when its extended for others, the check prefix can be updated.

File: llvm/docs/CommandGuide/lit.rst
Comment (arichardson) [inline llvm/docs/CommandGuide/lit.rst:538]: I see this is consistent with the already existing basename_t, but I feel that it would make a lot more sense if this was consistent with `%{s:real}` and `%{/s:regex_replacement}` as below, i.e. `%{s:basename}`.

Comment (jurahul) [reply llvm/docs/CommandGuide/lit.rst:538]: Let me look into changing this. I agree this makes sense.

Comment (jurahul) [reply llvm/docs/CommandGuide/lit.rst:538]: I created: https://github.com/llvm/llvm-project/pull/111062, will start review once checks pass.

**Pull request merged**
